texar/torch/modules/tokenizers/pretrained_bert_tokenizer_utils.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utils of pre-trained BERT tokenizer.
+"""
+
+from typing import Dict, List, Optional
+
+import collections
+import unicodedata
+
+
+__all__ = [
+    "load_vocab",
+    "BasicTokenizer",
+    "WordpieceTokenizer",
+]
+
+
+def load_vocab(vocab_file: str) -> Dict[str, int]:
```

Comment:
`tx.data.Vocabulary` currently has limited functionalities, for example, it only considers four types of special tokens (`bos`, `eos`, `pad`, `unk`); it does not allow special tokens in the `vocab_file` when you `load` the `vocab_file`... I feel that we cannot directly apply `tx.data.Vocabulary` in `PretrainedTokenizer` right now.

Reply:
Can you assess the possibility of upgrading `tx.data.Vocab` to support these functionalities? It'd be good (and ultimately necessary) to have one common Vocab class (instead of many).

Reply:
Such code refactoring will somehow influence the internal use of `Vocab` in our `data` module. I need to read that part thoroughly to start working on the code refactoring. But I feel that such refactor should not be very complicated. I can open an issue to track this issue at first.